tool/tsh/options.go

```diff
@@ -163,14 +136,126 @@ type Options struct {
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	o.AddKeysToAgent = parsedValue
+	return nil
+}
+
+func setAgentForwardingModeOption(o *Options, val string) error {
+	switch strings.ToLower(val) {
+	case "no":
+		o.ForwardAgent = client.ForwardAgentNo
+	case "yes":
+		o.ForwardAgent = client.ForwardAgentYes
+	case "local":
+		o.ForwardAgent = client.ForwardAgentLocal
+	default:
+		return trace.BadParameter("invalid agent forwarding mode: %s", val)
+	}
+	return nil
+}
+
+func setForwardX11Option(o *Options, val string) error {
+	parsedValue, err := parseBoolOption(val)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	o.ForwardX11 = parsedValue
+	return nil
+}
+
+func setForwardX11TimeoutOption(o *Options, val string) error {
+	parsedValue, err := parseUintOption(val)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	o.ForwardX11Timeout = parsedValue
+	return nil
+}
+
+func setForwardX11TrustedOption(o *Options, val string) error {
+	parsedValue, err := parseBoolOption(val)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	o.ForwardX11Trusted = parsedValue
+	return nil
+}
+
+func setRequestTTYOption(o *Options, val string) error {
+	parsedValue, err := parseBoolOption(val)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	o.RequestTTY = parsedValue
+	return nil
+}
+
+func setStrictHostKeyCheckingOption(o *Options, val string) error {
+	parsedValue, err := parseBoolOption(val)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	o.StrictHostKeyChecking = parsedValue
+	return nil
+}
+
+func parseBoolOption(val string) (bool, error) {
+	if val != "yes" && val != "no" {
+		return false, trace.BadParameter("invalid bool option value: %s", val)
+	}
+	return utils.AsBool(val), nil
+}
+
+func parseBoolTrueOption(val string) (bool, error) {
+	if val != "yes" {
+		return false, trace.BadParameter("invalid true-only bool option value: %s", val)
+	}
+	return utils.AsBool(val), nil
+}
+
+func parseUintOption(val string) (uint, error) {
+	valUint, err := strconv.ParseUint(val, 10, 0)
+	if err != nil {
+		return 0, trace.BadParameter("invalid uint option value: %s", val)
+	}
+	return uint(valUint), nil
 }
 
 func parseOptions(opts []string) (Options, error) {
-	// By default, Teleport prefers strict host key checking and adding keys
-	// to system SSH agent.
 	options := Options{
+		// By default, Teleport prefers strict host key checking and adding keys
+		// to system SSH agent.
 		StrictHostKeyChecking: true,
 		AddKeysToAgent:        true,
+		// As in OpenSSH, untrusted mode is the default unless explicitly set to false.
+		// Although it makes clients using x11 forwarding vulnerable to some XServer
+		// related attacks (such as Keystroke monitoring), most programs do not properly
+		// support untrusted mode and will crash. For these reasons, users are encouraged
+		// to enable x11 forwarding with caution, as an attacker that can bypass file
+		// permissions on the remote host may be able to carry out such attacks.
+		ForwardX11Trusted: true,
```

Comment:
Can we make sure this is covered in our docs as well? Seems too important to just be buried in the code.